mono/mini/mini-exceptions.c

```diff
@@ -3179,6 +3135,73 @@ mono_handle_native_crash (const char *signal, void *ctx, MONO_SIG_HANDLER_INFO_T
 	mono_runtime_printf_err ("\nNo native Android stacktrace (see debuggerd output).\n");
 #endif
 #endif
+}
+void
+mono_dump_native_crash_info (const char *signal, void *ctx, MONO_SIG_HANDLER_INFO_TYPE *info)
+{
+	print_process_map ();
+
+	dump_memory_around_ip (ctx);
+
+	dump_native_stacktrace ();
+}
+
+static void
+mono_post_native_crash_handler (const char *signal, void *ctx, MONO_SIG_HANDLER_INFO_TYPE *info, gboolean crash_chaining)
+{
+	if (!crash_chaining) {
+		/*Android abort is a fluke, it doesn't abort, it triggers another segv. */
+#if defined (HOST_ANDROID)
+		exit (-1);
+#else
+		abort ();
+#endif
+	}
+}
+#else
+extern void
+mono_dump_native_crash_info (const char *signal, void *ctx, MONO_SIG_HANDLER_INFO_TYPE *info);
```

Comment:
There's no reason why we can't keep all implementations symmetric and move the unix code to mini-posix.

Reply:
Will do, doing same thing as done to mini-windows.c, move to mini-posix.c?

Reply:
Yes, we don't need to have an impl or a forward decl in a source file.